File: backend/app/schemas/comunicados.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.models.enums import StatusComunicado, TipoDestinatarioComunicado


class DestinatarioRef(BaseModel):
    tipo: TipoDestinatarioComunicado
    id: UUID


class ComunicadoListItem(BaseModel):
    id: UUID
    titulo: str
    status: StatusComunicado
    publicado_em: datetime | None = None
    lido: bool = False
    preview_corpo: str | None = None


class ComunicadoLeituraItem(BaseModel):
    usuario_id: UUID
    nome_exibicao: str
    lido: bool
    lido_em: datetime | None = None


class ComunicadoLeiturasResponse(BaseModel):
    total_destinatarios: int
    total_lidos: int
    itens: list[ComunicadoLeituraItem] = []


class ComunicadoDetail(BaseModel):
    id: UUID
    titulo: str
    corpo: str
    status: StatusComunicado
    publicado_em: datetime | None = None
    imagens_urls: list[str] = []
    destinatarios: list[DestinatarioRef] = []
    lido: bool = False
    total_destinatarios: int | None = None
    total_lidos: int | None = None


class ComunicadoCreate(BaseModel):
    titulo: str = Field(min_length=1)
    corpo: str = ""
    imagens_urls: list[str] = []
    destinatarios: list[DestinatarioRef] = []
    status_inicial: StatusComunicado = StatusComunicado.rascunho


class ComunicadoPatch(BaseModel):
    titulo: str | None = None
    corpo: str | None = None
    imagens_urls: list[str] | None = None
    destinatarios: list[DestinatarioRef] | None = None

File: frontend/lib/api/dtos/comunicados.ts
import type { StatusComunicado, TipoDestinatarioComunicado } from "./common"

export interface DestinatarioRef {
  tipo: TipoDestinatarioComunicado
  id: string
}

export interface ComunicadoListItem {
  id: string
  titulo: string
  status: StatusComunicado
  publicado_em?: string | null
  lido: boolean
  preview_corpo?: string | null
}

export interface ComunicadoLeituraItem {
  usuario_id: string
  nome_exibicao: string
  lido: boolean
  lido_em?: string | null
}

export interface ComunicadoLeiturasResponse {
  total_destinatarios: number
  total_lidos: number
  itens: ComunicadoLeituraItem[]
}

export interface ComunicadoDetail {
  id: string
  titulo: string
  corpo: string
  status: StatusComunicado
  publicado_em?: string | null
  imagens_urls: string[]
  destinatarios: DestinatarioRef[]
  lido: boolean
  total_destinatarios?: number | null
  total_lidos?: number | null
}

export interface ComunicadoCreate {
  titulo: string
  corpo?: string
  imagens_urls?: string[]
  destinatarios?: DestinatarioRef[]
  status_inicial?: StatusComunicado
}

export interface ComunicadoPatch {
  titulo?: string | null
  corpo?: string | null
  imagens_urls?: string[] | null
  destinatarios?: DestinatarioRef[] | null
}
